subsys/logging/log_output.c

```diff
@@ -506,6 +518,13 @@ void log_output_msg_process(const struct log_output *log_output,
 	bool raw_string = (level == LOG_LEVEL_INTERNAL_RAW_STRING);
 	int prefix_offset;
 
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+	if (flags & LOG_OUTPUT_FLAG_FORMAT_SYST) {
```

Comment:
i cases like that (runtime + compile time condition) i prefer to create local function (e.g. `is_format_syst_enabled(flags)`) which simply wraps `IS_ENABLED(SYST_ENABLE) && (flags & SYST_FLAG)`. that reduces ifdefs to 0 (no need to ifdef function declarations too)

Reply:
Thanks